src/aosm/azext_aosm/delete/delete.py

```diff
@@ -0,0 +1,330 @@
+        try:
+            poller = self.api_clients.aosm_client.network_function_definition_versions.begin_delete(
+                resource_group_name=self.config.publisher_resource_group_name,
+                publisher_name=self.config.publisher_name,
+                network_function_definition_group_name=self.config.nfdg_name,
+                network_function_definition_version_name=self.config.version,
+            )
+            LongRunningOperation(self.cli_ctx, "Deleting NFDV...")(poller)
+            print("Deleted NFDV.")
+        except Exception:
+            logger.error(
+                "Failed to delete NFDV %s from group %s",
+                self.config.version,
+                self.config.nfdg_name,
+            )
+            raise
+
+    def delete_nsdv(self):
+        assert isinstance(self.config, NSConfiguration)
+        message = (
+            f"Delete NSDV {self.config.nsd_version} from group"
+            f" {self.config.nsd_name} and publisher {self.config.publisher_name}"
+        )
+        logger.debug(message)
+        print(message)
+        try:
+            poller = self.api_clients.aosm_client.network_service_design_versions.begin_delete(
+                resource_group_name=self.config.publisher_resource_group_name,
+                publisher_name=self.config.publisher_name,
+                network_service_design_group_name=self.config.nsd_name,
+                network_service_design_version_name=self.config.nsd_version,
+            )
+            LongRunningOperation(self.cli_ctx, "Deleting NSDV...")(poller)
+            print("Deleted NSDV.")
+        except Exception:
+            logger.error(
+                "Failed to delete NSDV %s from group %s",
+                self.config.nsd_version,
+                self.config.nsd_name,
+            )
+            raise
+
+    def delete_artifact_manifest(self, store_type: str) -> None:
+        """
+        _summary_
+
+        :param store_type: "sa" or "acr"
+        :raises CLIInternalError: If called with any other store type         :raises
+                Exception if delete throws an exception
+        """
+        if store_type == "sa":
+            assert isinstance(self.config, VNFConfiguration)
+            store_name = self.config.blob_artifact_store_name
+            manifest_names = [self.config.sa_manifest_name]
+        elif store_type == "acr":
+            store_name = self.config.acr_artifact_store_name
+            manifest_names = self.config.acr_manifest_names
+        else:
+            from azure.cli.core.azclierror import CLIInternalError
+
+            raise CLIInternalError(
+                "Delete artifact manifest called for invalid store type. Valid types"
+                " are sa and acr."
+            )
+
+        for manifest_name in manifest_names:
+            message = f"Delete Artifact manifest {manifest_name} from artifact store {store_name}"
+            logger.debug(message)
+            print(message)
```

Comment:
Same as https://github.com/Azure/azure-cli-extensions/pull/6426/files#r1366420334

Reply:
We need to rethink how we output to console. These changes are too widespread to make before GA release, so I've opened work item https://dev.azure.com/msazuredev/AzureForOperators/_workitems/edit/992775 to track.